addons/missile_defense/functions/fnc_createSystem.sqf

```diff
@@ -0,0 +1,54 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Brett Mayson
+ * Create a missile defense system.
+ *
+ * Arguments:
+ * 0: Sides <ARRAY of SIDEs> - The sides that the missile defense system will engage. If not provided, it will engage all sides.
+ * 1: ID <STRING> - Optional, The ID of the missile defense system to create.
+ *
+ * Return Value:
+ * <STRING> - The ID of the created missile defense system.
+ *
+ * Example:
+ * [] call ace_missile_defense_createSystem
+ *
+ * Public: No
+ */
+
+if (!isServer) exitWith {
+    ERROR("missile_defense functions only run on server");
+};
+
+private _id = if (count _this > 1) then {
+    _this select 1
+} else {
+    format ["%1", diag_tickTime]
```

Comment:
This will fail when you will try to register multiple systems in same second/frame.